sim/dex-data.ts

```diff
@@ -662,75 +602,66 @@ export class Template extends BasicEffect {
 	eventPokemon?: EventInfo[] ;
 	/** True if a pokemon is mega. */
 	isMega?: boolean;
+	/** True if a pokemon is primal. */
+	isPrimal?: boolean;
 	/** True if a pokemon is a forme that is only accessible in battle. */
 	battleOnly?: boolean;
 
-	constructor(data: AnyObject, moreData: AnyObject | null = null, moreData2: AnyObject | null = null, moreData3: AnyObject | null = null) {
-		super(data, moreData);
-		if (moreData2) Object.assign(this, moreData2);
-		if (moreData3) Object.assign(this, moreData3);
-		// @ts-ignore
-		this.fullname = 'pokemon: ' + this.name;
+	constructor(data: AnyObject, moreData: AnyObject | null = null, ...evenMoreData: (AnyObject | null)[]) {
+		data = combine(data, moreData);
+		super(data);
+		data = combine(data, ...evenMoreData);
```

Comment:
this is super weird - i'd ideally like to do away with `...evenMoreData` and just have one `...moreData` param encompassing everything, but I tried to match the original as much as possible to begin with.

Reply:
I'm not sure what the point of matching the original is here?